### PR TITLE
separate camelcase logic from space processing logic

### PR DESCRIPTION
**Pull request by chris-ha458**
separate long space removal and camel case isolation logic
add preprocessing only pretokenizer removal logic
add separate argument for camelcase

### CODE REVIEW

File: megatron/logging.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 # limitations under the License.
 
 import sys
+
```
Comment (Quentin-Anthony) [inline megatron/logging.py:16]: remove this newline

File: megatron/model/flash_attention.py [hunk 2]
```diff
@@ -8,6 +8,12 @@
 import flash_attn_cuda
 
 
+def flash_attn_unpadded_unpacked_func_triton(
```
Comment (Quentin-Anthony) [inline megatron/model/flash_attention.py:11]: Where are all these flash attention changes coming from?

File: megatron/model/transformer.py [hunk 3]
```diff
@@ -278,17 +278,13 @@ def __init__(
...
 
-                if self.pos_emb == "alibi":
-                    self.flash_attention_function = (
-                        flash_attn_unpadded_qkvpacked_func_triton
-                    )
-                else:
-                    self.flash_attention_function = (
-                        flash_attn_unpadded_qkvpacked_func_cuda
-                    )
+                self.flash_triton_fn = flash_attn_unpadded_unpacked_func_triton
```
Comment (Quentin-Anthony) [inline megatron/model/transformer.py:285]: I believe your branch is based on an older version of gpt-neox. Please rebase your PR to the latest `main` branch

Comment (chris-ha458) [reply megatron/model/transformer.py:285]: sorry about that. currently the polyglot branch maintainer is refactoring the branch for tokenizer inclusion.

Comment (StellaAthena) [reply megatron/model/transformer.py:285]: @chris-ha458 Would it make sense to close this PR and re-open it once the branch has been updated then?

**Comment by StellaAthena** [comment]
Closing while the branch is rebased to get inline with `main`

**Pull request closed without merging**